elastic/beats
# Fix container enricher with correct container ids for pods with multiple containers

## What does this PR do?



This PR fixes a bug in NewContainerMetadataEnricher used by Kubernetes module's container metricset.
In case a pod consists of more than one container, the ecs field `container.id` was the same for all containers.
This was happening because the `meta` struct used for both containers was the same and the `container.id` value was getting overwritten.


Also this PR aligns `kubernetes.container.id` and `container.id` of `state_container` metricset.
So far `kubernetes.container.id` was not splitted and contained both the runtime and the id (`containerd://99781780c8d2eaa7a0333e158c49969b20cdf067f7b00d8b93022b508ee09724`)
On the other hand `container.id` was splitted to `99781780c8d2eaa7a0333e158c49969b20cdf067f7b00d8b93022b508ee09724` leading to confusion.
Now on it will be splitted in all cases.

## Why is it important?



This was leading to errors and containers not found. 
Issue https://github.com/elastic/obs-infraobs-team/issues/935

## Checklist





## How to test this PR locally


To test this locally just create an elastic-stack and deploy default kubernetes [metricbeat](https://github.com/elastic/beats/blob/main/deploy/kubernetes/metricbeat-kubernetes.yaml). 
Deploy a pod in the k8s cluster with two containers and then in Kibana for that specific pod.name and container metricset, observe the container.ids of the different container names of that pod. 

## Issues

- Closes https://github.com/elastic/beats/issues/34533

## Screenshots



Before the fix:

<img width="1340" alt="before_error" src="https://user-images.githubusercontent.com/26270880/217490381-c104e636-6fa7-4f79-9305-b6487dcf3806.png">


After the fix:

<img width="1530" alt="new fixed" src="https://user-images.githubusercontent.com/26270880/217490426-b101e723-fd25-4e88-8d64-15c46443fa04.png">



## Logs
